bundles/org.openhab.binding.edf/src/main/java/org/openhab/binding/edf/internal/handler/TariffHandler.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.edf.internal.handler;
+
+import static org.openhab.binding.edf.internal.EDFBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.ws.rs.HttpMethod;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.edf.internal.dto.Tariff;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.CurrencyUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TariffHandler} is the base class for Tariff Things. It takes care of
+ * update logic and update scheduling once a day.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public abstract class TariffHandler<T extends Tariff> extends BaseThingHandler {
+    private static final String URL = "https://www.data.gouv.fr/fr/datasets/r/%s";
+    private static final int REFRESH_FIRST_HOUR_OF_DAY = 0;
+    private static final int REFRESH_FIRST_MINUTE_OF_DAY = 1;
+
+    private final Logger logger = LoggerFactory.getLogger(TariffHandler.class);
+    private final String url;
+    protected final List<T> tariffs = new ArrayList<>();
+
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private @Nullable String fileCache = null;
+    private int puissance = 6;
+
+    public TariffHandler(Thing thing, String dataset) {
+        super(thing);
+        this.url = URL.formatted(dataset);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        Object puiss = getConfig().get("puissance");
+        puissance = puiss != null ? ((BigDecimal) puiss).intValue() : 6;
+        scheduler.execute(this::updateData);
```

Comment:
This is not done in most of the bindings I [looked](https://github.com/search?q=repo%3Aopenhab%2Fopenhab-addons%20initialize%20scheduler.execute%20this&type=code) at.

Reply:
That's a pitty because it is probably the reason why some users notice log messages saying that stuff is done while thing handler is already disposed.

Reply:
I think it is important, especially while `updateData` is itself scheduling a job.